modules/hitlnext/src/views/full/app/components/LiveChat.tsx

```diff
@@ -11,32 +11,59 @@ interface Props {
   currentAgent: IAgent
 }
 
+const INJECTION_ID = 'bp-channel-web-injection'
+const INJECTION_URL = 'assets/modules/channel-web/inject.js'
 const WEBCHAT_ID = 'hitl-webchat'
 const WRAPPER_ID = `${WEBCHAT_ID}-wrapper`
+let intervalId
 
 const LiveChat: React.FC<Props> = ({ handoff, currentAgent }) => {
+  const [webchatLoaded, setWebchatLoaded] = useState(false)
   const [webchatReady, setWebchatReady] = useState(false)
 
-  function getWebchatStore() {
-    return window[WEBCHAT_ID].webchat_store
+  const loadWebchat = () => {
+    if (document.getElementById(INJECTION_ID)) {
+      setWebchatLoaded(true)
+      return
+    }
+
+    const script = window.document.createElement('script')
+    script.src = INJECTION_URL
+    script.id = INJECTION_ID
+    window.document.body.appendChild(script)
+
+    intervalId = setInterval(() => {
+      if (window.botpressWebChat) {
```

Comment:
We must ensure that the webchat is loaded only once, but even if the script tag is added, it doesn't means that the webchat is ready. The first check ensures that the script is loaded, and the second ensures that it has loaded completely. Since the method is only called once, there should not be any issues

Reply:
Oh okay, I see! Thanks for the clarification. Can you please add your response as a comment in the code so that it will also be clear for everyone?